cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGeneratortest.scala

```diff
@@ -0,0 +1,786 @@
+          | """.stripMargin.trim)
+    }
+
+    //24 //TODO: parseUnnest
+    "translate a text contain + time + geo id set filter and group by hashtags" in {
+      ok
+    }
+
+    //25
+    "translate a simple filter by time and group by time query min id" in {
+      val filter = Seq(timeFilter)
+      val group = GroupStatement(Seq(byHour), Seq(aggrMin))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select hour(t.`create_at`) as `hour`,min(t.`id`) as `min`
+          |from `twitter_ds_tweet` t
+          |where t.`create_at` >= '2016-01-01 00:00:00.000' and t.`create_at` < '2016-12-01 00:00:00.000'
+          |group by `hour`
+          | """.stripMargin.trim)
+    }
+
+    //26
+    "translate a simple filter by time and group by time query sum id" in {
+      val filter = Seq(timeFilter)
+      val group = GroupStatement(Seq(byHour), Seq(aggrSum))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select hour(t.`create_at`) as `hour`,sum(t.`id`) as `sum`
+          |from `twitter_ds_tweet` t
+          |where t.`create_at` >= '2016-01-01 00:00:00.000' and t.`create_at` < '2016-12-01 00:00:00.000'
+          |group by `hour`
+          |  """.stripMargin.trim)
+    }
+
+    //27
+    "translate a simple filter by time and group by time query avg id" in {
+      val filter = Seq(timeFilter)
+      val group = GroupStatement(Seq(byHour), Seq(aggrAvg))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select hour(t.`create_at`) as `hour`,avg(t.`id`) as `avg`
+          |from `twitter_ds_tweet` t
+          |where t.`create_at` >= '2016-01-01 00:00:00.000' and t.`create_at` < '2016-12-01 00:00:00.000'
+          |group by `hour`
+          | """.stripMargin.trim)
+    }
+
+    //28
+    "translate a text contain filter and group by geocell 10th" in {
+      val filter = Seq(textFilter)
+      val group = GroupStatement(Seq(byGeocell10), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select st_astext(Point(truncate(st_x(t.`coordinate`),1),truncate(st_y(t.`coordinate`),1)))  as `cell`,count(*) as `count`
```

Comment:
st_astext: print or return the geometric results as text, e.g.: "POINT(1, 2)",
without this function, the query would return a messed code.